scripts/home_allocation.py

```diff
@@ -0,0 +1,138 @@
+import argparse
+import os
+from typing import Literal
+
+import geopandas as gpd
+import polars as pl
+from context import nhs
+from fiona.drvsupport import supported_drivers
+from loguru import logger
+
+from nhs.config import simulation_config
+
+geography = nhs.data.geography
+read_parquet = nhs.data.read_parquet
+join_coords_with_area = nhs.data.join_coords_with_area
+to_geo_dataframe = nhs.data.to_geo_dataframe
+read_shapefile = nhs.data.read_shapefile
+join_coords_with_area = nhs.data.join_coords_with_area
+join_census_with_coords = nhs.data.join_census_with_coords
+randomly_assign_census_features = nhs.data.randomly_assign_census_features
+sample_census_feature = nhs.data.sample_census_feature
+
+
+def main(
+    config_path: str,
+    gnaf_fname: str,
+    census_fname: str,
+    out_fname: str,
+    strategy: Literal["join_nearest", "filter"] | None,
+) -> None:
+    # Required for fiona - reads shapefiles
+    supported_drivers["ESRI Shapefile"] = "rw"
+
+    try:
+        data_config = nhs.config.data_config(config_path)
+        logger_config = nhs.config.logger_config(config_path)
+        simulation_config = nhs.config.simulation_config(config_path)
+    except Exception as e:
+        logger.critical(
+            f"Failed to load configuration at {config_path} with exception {e}, terminating..."
+        )
+        exit(1)
+
+    logger.enable("nhs")
+    nhs.logging.config_logger(logger_config)
+
+    gnaf_path = os.path.join(data_config["gnaf_path"], gnaf_fname)
```

Comment:
Is this a limitation only being able to read in one GNAF component? We could easily get it into one big file if that makes more sense.

Reply:
yup, all that's needed is that the lazyframe have the required columns detailed in the function docstring